integration/spark/app/src/main/java/io/openlineage/spark/agent/lifecycle/ContextFactory.java

```diff
@@ -26,9 +27,11 @@ public class ContextFactory {
 
   public final EventEmitter openLineageEventEmitter;
   private final OpenLineageEventHandlerFactory handlerFactory;
+  private final OpenLineageYaml openLineageYaml;
 
-  public ContextFactory(EventEmitter openLineageEventEmitter) {
+  public ContextFactory(EventEmitter openLineageEventEmitter, OpenLineageYaml openLineageYaml) {
     this.openLineageEventEmitter = openLineageEventEmitter;
+    this.openLineageYaml = openLineageYaml;
```

Comment:
Can't we pass direct object that we're using later rather than whole `OpenLineageYaml`?

Reply:
I think it's better to pass a whole config to the context for future usecases (for example config based dataset facet builder). The problem is that sometimes for Spark we rely on Spark conf entries, which means that if you put config entry to openlineage.yaml, it will not work.